packages/data-context/src/data/ProjectConfigIpc.ts

```diff
@@ -238,6 +252,59 @@ export class ProjectConfigIpc extends EventEmitter {
 
     debug('fork child process %o', { CHILD_PROCESS_FILE_PATH, configProcessArgs, childOptions: _.omit(childOptions, 'env') })
 
+    let isProjectUsingESModules = false
+
+    try {
+      const pkgJson = fs.readJsonSync(path.join(this.projectRoot, 'package.json'))
+
+      isProjectUsingESModules = pkgJson.type === 'module'
+    } catch (e) {
+      // project does not have `package.json` or it was not found
+      // reasonable to assume not using es modules
+    }
+
+    if (!childOptions.env) {
+      childOptions.env = {}
+    }
+
+    // If they've got TypeScript installed, we can use
+    // ts-node for CommonJS
+    // ts-node/esm for ESM
+    if (this.hasTypeScriptInstalled()) {
```

Comment:
This function is pretty similar to what we have above in `MigrationActions.ts`. Is there any way we could share the code? Is using ESModules not a possibility in `MigrationActions.ts`?

Reply:
I thought about this a bit more and 9.x did **not** support ES Modules, and the plugin resolution needs to match exactly what 9.x did to work the same - I see some potential risk in sharing this code, since it means we'd be doing something slightly differently and there might be some edge cases around migrating projects we haven't considered.